crates/turborepo-lib/src/run/scope/change_detector.rs

```diff
@@ -202,3 +216,58 @@ impl From<ChangedPackagesError> for ChangeDetectError {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use test_case::test_case;
+
+    use super::{ChangeDetectError, SCMChangeDetector};
+
+    #[cfg(unix)]
+    #[test_case("/a/b/c", &["package.lock"], "/a/b/c/package.lock", Ok(true) ; "simple")]
+    #[test_case("/a/b/c", &["a", "b", "c"], "/a/b/c/package.lock", Ok(false) ; "lockfile unchanged")]
+    #[test_case("/a/b/c", &["package.lock"], "/a/b/outside-repo/package.json", Err(ChangeDetectError::LockfileNotInRepo) ; "different file")]
+    fn test_lockfile_changed(
+        turbo_root: &str,
+        changed_files: &[&str],
+        lockfile_path: &str,
+        expected: Result<bool, ChangeDetectError>,
+    ) {
+        let turbo_root = turbopath::AbsoluteSystemPathBuf::new(turbo_root).unwrap();
+        let lockfile_path = turbopath::AbsoluteSystemPathBuf::new(lockfile_path).unwrap();
+        let changed_files = changed_files
+            .iter()
+            .map(|s| turbopath::AnchoredSystemPathBuf::from_raw(s).unwrap())
+            .collect();
+        let changes =
+            SCMChangeDetector::lockfile_changed(&turbo_root, &changed_files, &lockfile_path);
+
+        // we don't want to implement PartialEq on the error type,
+        // so simply compare the debug representations
+        assert_eq!(format!("{:?}", changes), format!("{:?}", expected));
```

Comment:
I always feel like there should be a better way to handle comparing possible errors. This might be the cleanest way around it I've seen, but it still feels slightly off.

Reply:
Yeah its a bit jank. I didn't want to just check that the enum variant was the same either...